src/magnify/apps/core/api/rooms.py

```diff
@@ -282,3 +161,90 @@ def meetings(self, request, pk=None):
             meetings, context={"request": request}, many=True
         )
         return response.Response(serializer.data, status=200)
+
+
+class RoomAdministratorListsModelMixin:
+    """List mixin for room access API."""
+
+    def list(self, request, *args, **kwargs):
+        """
+        Limit listed accesses to the ones related to a room for which the authenticated
+        user is an administrator or the owner.
+        """
+        user = self.request.user
+
+        if not user.is_authenticated:
+            raise PermissionDenied()
+
+        queryset = (
+            self.filter_queryset(self.get_queryset())
+            .filter(
+                Q(
+                    room__user_accesses__user=user,
+                    room__user_accesses__role__in=[
+                        models.UserRoleChoices.ADMIN,
+                        models.UserRoleChoices.OWNER,
+                    ],
+                )
+                | Q(
+                    room__group_accesses__group__members=user,
+                    room__group_accesses__role=models.GroupRoleChoices.ADMIN,
+                )
+            )
+            .distinct()
+        )
+
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+
+        serializer = self.get_serializer(queryset, many=True)
+        return response.Response(serializer.data)
```

Comment:
Is the reimplementation of the list viewset mandatory? I'm may have missed something, but why don't rely on the DRF methods directly? If you want a mixin:
``` python
class RoomAdministratorListsModelMixin:
    """List mixin for room access API."""

    def get_permissions(self):
        """User only needs to be authenticated to list rooms access"""
        if self.action in ["list"]:
            permission_classes = [UserIsAuthenticated]
        else:
            return super().get_permissions()

        return [permission() for permission in permission_classes]

    def get_queryset(self):
        """Return the queryset according to the action."""
        queryset = super().get_queryset()
        if self.action in ["list""]:
            queryset = (
                .filter(
                    Q(
                        room__user_accesses__user=user,
                        room__user_accesses__role__in=[
                            models.UserRoleChoices.ADMIN,
                            models.UserRoleChoices.OWNER,
                        ],
                    )
                    | Q(
                        room__group_accesses__group__members=user,
                        room__group_accesses__role=models.GroupRoleChoices.ADMIN,
                    )
                )
                .distinct()
            )
        return queryset
```

I'm not sure it would work, but it keeps the concerns separation

``` python
class RoomUserAccessViewSet(
    RoomAdministratorListsModelMixin,
    ModelViewSet,
):
    ...
```

Reply:
Oh wow... all tests are passing so I'm pretty confident that it works! Much nicer split of responsibility yes! :medal_sports:

Reply:
Added a fixup: https://github.com/openfun/jitsi-magnify/pull/132/commits/eac907f402f55f213c50e9b2e3f64d0ecd6d1617